bundles/org.openhab.binding.anel/src/test/java/org/openhab/binding/anel/internal/AnelUdpConnectorTest.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.anel.internal;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+import java.util.LinkedHashSet;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.openhab.binding.anel.internal.auth.AnelAuthentication;
+import org.openhab.binding.anel.internal.auth.AnelAuthentication.AuthMethod;
+
+/**
+ * This test requires a physical Anel device!
+ *
+ * @author Patrick Koenemann - Initial contribution
+ */
+@NonNullByDefault
+@Disabled // requires a physically available device in the local network
+public class AnelUdpConnectorTest {
+
+    /*
+     * The IP and ports for the Anel device under test.
+     */
+    private static final String HOST = "192.168.6.63"; // 63 / 64
+    private static final int PORT_SEND = 7500; // 7500 / 75001
+    private static final int PORT_RECEIVE = 7700; // 7700 / 7701
+    private static final String USER = "user7";
+    private static final String PASSWORD = "anel";
+
+    /* The device may have an internal delay of 200ms, plus network latency! Should not be <1sec. */
+    private static final int WAIT_FOR_DEVICE_RESPONSE_MS = 1000;
+
+    private static final ExecutorService EXECUTOR_SERVICE = Executors.newCachedThreadPool();
+
+    private static final Queue<String> receivedMessages = new ConcurrentLinkedQueue<>();
+
+    @Nullable
+    private static AnelUdpConnector connector;
+
+    @BeforeAll
+    public static void prepareConnector() {
+        connector = new AnelUdpConnector(HOST, PORT_RECEIVE, PORT_SEND, EXECUTOR_SERVICE);
+    }
+
+    @AfterAll
+    @SuppressWarnings("null")
+    public static void closeConnection() {
+        connector.disconnect();
+    }
+
+    @BeforeEach
+    @SuppressWarnings("null")
+    public void connectIfNotYetConnected() throws Exception {
+        Thread.sleep(100);
+        receivedMessages.clear(); // clear all previously received messages
```

Comment:
Sleeps won't work reliably in tests running on a high load CI machine. You can extend the test class with `JavaTest` and use `waitForAssertion()`.

Reply:
This test is disabled for CI because it requires a physical device. This is why it is probably always started manually in the IDE.